R/calculate-grades.R

```diff
@@ -1,223 +1,148 @@
-#' Aggregate assignment scores
+#' Calculate Grades
+#' This function calculates all grades based on the policy file.
+#'
+#' @param gs A processed data in wide format 
+#' @param policy A policy file
+#'
+#' @return A data frame
+#'
+#' @importFrom dplyr select
+#' 
+#' @export
+
+get_grades <- function(gs, policy){
+  
+  # flatten policy file
+  policy <- flatten_policy(policy)
+  
+  # convert gs into a matrix with only assignment info
+  grades_mat <- gs |>
+    #only assignment info
+    select(-get_id_cols(gs)) |> 
+    #convert lateness into minutes
+    mutate_at(vars(ends_with(" - Lateness (H:M:S)")), convert_to_min) |> 
+    #convert to matrix
+    data.matrix()
+  #rownames are SID of student
+  rownames(grades_mat) <- gs$SID
+  #pre_allotted cols
+  categories <- unlist(map(policy$categories, "category"))
+  empty <- matrix(nrow = length(gs$SID), ncol = length(categories))
+  colnames(empty) <- categories
+  grades_mat <- cbind(grades_mat, empty)
+  
+  
+  #iterate through each policy item
+  for (policy_item in policy$categories){
+    #compute calculations for each policy file and save into grades matrix
+    grades_mat <- get_category_grade(grades_mat, policy_item)
```

Comment:
Should `get_category_grade()` be renamed something like `add_columns()`?

Reply:
It should definitely have a different name, but add_columns doesn't seem accurate since it's just updating an existing column of NAs, right?